### DISCUSSION TITLE
Is it normal that DCM modeling tajes too much time?

### DISCUSSION THREAD

**Discussion by daniell-00** [bachlab/PsPM#556] [Q&A] [answered]
I Have tried to run dcm modeling with pspm_sf function on a dataset with 2 epochs of 5 min but it takes too much time, I quit when an hour have passed since the model started to run. Any advice?

% This is the script:
epochs = 'eventos_1.mat';
import = {...
    struct('type', 'scr', 'channel', 5, 'transfer', 'none'), ...
    struct('type', 'marker') ...
    };

for sujeto = 1
    options = struct('overwrite', true);
    pspm_import(sprintf('%i.edf',sujeto), 'edf', import, options);
    pspm_downsample('pspm_%i.mat', 10)
    model = struct('datafile', sprintf('pspm_%i.mat', sujeto), 'modelfile', sprintf('msf_%i', sujeto), 'timing', epochs, 'timeunits', 'seconds', 'method', 'dcm');
    pspm_sf(model, options);
end

% Thanks in advance

**Comment by teddphil** [comment, marked as answer]
Dear @daniell-00 
Thank you for posting the issue and adding the very detailed script.
Would you mind sending us the file so that we can replicate the issue and help resolve it?
My email address is skgtdz0@ucl.ac.uk. We will not share the datafile to any others.
If this is not possible, could you please give some brief description of the datafile so that we can try to give some suggestions?
Hope it helps.
Best wishes
Teddy
